### ISSUE TITLE
Poor contrast ratio for selected/active items in OptionList and ActionList

### ISSUE THREAD

**Issue by clauderic** [Shopify/polaris#1858]
## Issue summary
The contrast ratio for highlighting selected/active items in `OptionList` and `ActionList` feels poor. There is no visual differentiation between hovered items and selected items.

## Expected behavior
Our color usage guidelines indicate that the color `Indigo` should be used to signify active/highlighted states:

<img width="452" alt="Screen Shot 2019-07-17 at 12 52 33 PM" src="https://user-images.githubusercontent.com/1416436/61395273-0770c280-a893-11e9-8296-fb5cf9285343.png">

Based on those guidelines, this is what I would expect selected/active items to look like:
<img width="838" alt="Screen Shot 2019-07-17 at 12 46 54 PM" src="https://user-images.githubusercontent.com/1416436/61395277-0b044980-a893-11e9-96d7-5a5bf4726afb.png">

## Actual behavior
<img width="842" alt="Screen Shot 2019-07-17 at 12 47 09 PM" src="https://user-images.githubusercontent.com/1416436/61395315-20797380-a893-11e9-919f-91a7e3a9f9ff.png">

**Comment by annemarieleger** [comment]
+1!

Currently working with this pattern and the contrast of a selected item feels too subtle, especially when mixed with an action list; it is super important for merchants to understand the difference between _actions_ (row and column actions) and _options_ (vertical alignment). 

![image](https://user-images.githubusercontent.com/31770030/64183520-02aea080-ce38-11e9-9184-b76edceb1a06.png)

**Comment by andrewmcgov** [comment]
+1 - we are using the action list item `active` state to show which item is selected in a Popover, and it a little unclear which item is selected with the current contrast ratio.
